plugins/modules/zos_job_submit.py

```diff
@@ -745,57 +750,91 @@ def run_module():
     except SubmitJCLError as e:
         module.fail_json(msg=repr(e), **result)
     if jobId is None or jobId == "":
-        result["job_id"] = jobId
+        result["job_id"] = ""
         module.fail_json(
             msg="JOB ID RETURNED IS None. PLEASE CHECK WHETHER THE JCL IS CORRECT.",
             **result
         )
 
     result["job_id"] = jobId
     duration = 0
-    if wait is True:
-        # calculate the job elapse time
+    if not wait:
+        wait_time_s = 10
+
+    # real time loop - will be used regardless of 'wait' to capture data
+    starttime = timer()
+    loopdone = False
+    foundissue = None
+    while not loopdone:
         try:
-            waitJob = query_jobs_status(module, jobId)
-            job_msg = waitJob[0].get("ret_code").get("msg")
-        except SubmitJCLError as e:
+            job_output_txt = job_output(job_id=jobId)
+        except IndexError:
+            pass
+        except Exception as e:
+            result["err_detail"] = "{1} {2}.\n".format(
+                "Error during job submission.  The output is:", job_output_txt or " "
+            )
             module.fail_json(msg=repr(e), **result)
-        # while (job_msg.startswith("CC") or job_msg.startswith("ABEND")) is False:
-        while not re.search(
-            "^(?:{0})".format("|".join(JOB_COMPLETION_MESSAGES)), job_msg
-        ):
-            sleep(1)
-            duration = duration + 1
-            waitJob = job_output(job_id=jobId)
-            job_msg = waitJob[0].get("ret_code").get("msg")
-            if re.search("^(?:{0})".format("|".join(JOB_COMPLETION_MESSAGES)), job_msg):
-                break
-            if duration == wait_time_s:  # Long running task. timeout return
-                break
 
-    try:
-        result = get_job_info(module, jobId, return_output)
+        if bool(job_output_txt):
+            jot_retcode = job_output_txt[0].get("ret_code")
+            if bool(jot_retcode):
+                job_msg = jot_retcode.get("msg")
+                if re.search(
+                    "^(?:{0})".format("|".join(JOB_COMPLETION_MESSAGES)), job_msg
+                ):
+                    loopdone = True
+                    # if the message doesn't have a CC, it is an improper completion (error/abend)
+                    if re.search("^(?:CC)", job_msg) is None:
+                        foundissue = job_msg
+
+        if not loopdone:
+            checktime = timer()
+            duration = round(checktime - starttime)
+            if duration >= wait_time_s:
+                loopdone = True
+                result["message"] = {
+                    "stdout": "Submit JCL operation succeeded but it is a long running job. Timeout is "
```

Comment:
@richp405 - question, when its a long job (wait=true & execution > wait_time_s)  is the job ID available to us such we could share with the user at this time such they could use it to poll, if not its not a big deal they have the Job name.
Second question, should we enhance this to suggest they use zos_job_query to poll for jobs that are long running? 

Ie `Submit JCL operation succeeded but is a long running job and exceeded the configured  wait_time_s=XX. Consider using module zos_job_query to poll for long running jobs.` 

Something like this?

Reply:
I'm not sure on the 0=forever, since there is a default value on the system (originally 1 second, but its a configurable parm).  I'd need to understand how we could justify freezing a playbook for that.  However, the second idea is doable, and should be relatively easy.  

There are 2 places where the delay is an issue... during the initial call, so that we don't have a Job id, and during the loop, where we have the id, but no terminator.  Adding job id to the latter cases should be easy.   I have seen delays in the later loops, but never in the initial call...usually the jobid returns sub-second, even if it's only queued.